### ISSUE TITLE
how to use vuex in Typescript project

### ISSUE THREAD

**Issue by langlangbody** [WingGao/mpvue-ts-demo#15]
use vuex-class ?

**Comment by linfan68** [comment]
vuex-class is one option if your project is small and the same store getter/state/action is not use in multiple places, since each PAGE that uses store needs to declare the type. To me, that's not "type-safe"

We're using this approach in our project: 
https://github.com/istrib/vuex-typescript/blob/master/src/index.ts

it heavily uses typescript type inferring and generics to generate type-safe callable functions/getters for you. The down side is it is very hard to understand how it works (with very few code though). Also I think it's pushing TS compiler too hard and we're seeing compiler crash with ts 2.8.x in some cases (but ts 3.0 works a lot better), not big deal and workaroundable, just give you an idea how complicated it is. 

Basically, my suggestion is take a look at the source code of vuex-typescript. If you get some idea about what it is doing, then probably it's the right choice for you.

**Comment by langlangbody** [comment]
@linfan68  thank you very much for your answer.
